plugins/jenkins/app/models/samson/jenkins.rb

```diff
@@ -30,17 +56,126 @@ def initialize(job_name, deploy)
       @deploy = deploy
     end
 
+    def jenkins_job_config
+      conf = Rails.cache.fetch(job_name + "_conf", expires_in: 1.day, race_condition_ttl: 5.minute) do
+        conf = client.job.get_config(job_name)
+      end
+      Nokogiri::XML(conf)
+    end
+
+    def configure_job_config
+      conf = jenkins_job_config
+      post_config = false
+      expected_build_parameters = JENKINS_BUILD_PARAMETERS.keys
+      present_build_params = get_build_params(conf).map(&:to_sym)
+      missing_build_params = expected_build_parameters.to_set - present_build_params.to_set
+      if missing_build_params.any?
+        add_build_parameters(conf, missing_build_params)
+        post_config = true
+      end
+      unless check_description(conf)
+        add_job_description(conf)
+        post_config = true
+      end
+      if post_config
+        post_job_config(conf)
+      end
+    end
+
+    def get_build_params(conf)
+      params = conf.xpath("//hudson.model.StringParameterDefinition")
+      params_array = []
+      params.each do |param|
+        param.children.each do |value|
+          if value.name == "name"
+            if val = value.content.split(JENKINS_BUILD_PARAMETRS_PREFIX, 2)[1]
+              params_array << val
+            end
+          end
+        end
+      end
+      params_array
+    end
+
+    def get_description(conf)
+      conf.xpath("//description").first
+    end
+
+    def add_job_description(conf)
```

Comment:
some examples / docs on what this is doing ... ideally links to docs on why names have leading `*` would be helpful for future acheologists